hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerKeyMapperTask.java

```diff
@@ -263,6 +268,22 @@ public Pair<String, Boolean> process(OMUpdateEventBatch events) {
     return new ImmutablePair<>(getTaskName(), true);
   }
 
+  /**
+   * Checks if the object passed is an instance of `OmKeyInfo`.
+   *
+   * @param obj The object to check.
+   * @return True if the object is an instance of `OmKeyInfo`, false otherwise.
+   */
+  private boolean isOmKeyInfo(Object obj) {
+    if (obj instanceof OmKeyInfo) {
+      return true;
+    } else {
+      LOG.warn("Unexpected value type {} for key. Skipping processing.",
+          obj.getClass().getName());
+      return false;
```

Comment:
@ArafatKhan2198 thanks for providing workaround, however as we know ideally keyTable and fileTable should not have RepeatedKeyInfo objects and events for these key types should not raised for these 2 tables. So as @smengcl suggested, may be you can add some validation logic in "`org.apache.hadoop.ozone.recon.tasks.OMDBUpdatesHandler#processEvent`" where you can maintain an enum or map for keyType class name with its corresponding tablename as reference data to validate against and while if validation fails, you can log as ERROR with details like keyType, tablename, event action (PUT, DELETE, UPDATE etc), and  not allow such events to process as well to avoid flowing downstream. This way we can stop such events there itself and change will be at one place instead of all classes.